core/lib/tee_verifier/src/lib.rs

```diff
@@ -191,36 +212,51 @@ fn map_log_tree(
     idx: &mut u64,
 ) -> anyhow::Result<TreeInstruction> {
     let key = storage_log.key.hashed_key_u256();
-    Ok(match (storage_log.is_write(), *tree_log_entry) {
-        (true, TreeLogEntry::Updated { leaf_index, .. }) => {
+    let tree_instruction = match (storage_log.is_write(), storage_log.kind, *tree_log_entry) {
+        (true, _, TreeLogEntry::Updated { leaf_index, .. }) => {
             TreeInstruction::write(key, leaf_index, H256(storage_log.value.into()))
         }
-        (true, TreeLogEntry::Inserted) => {
+        (true, _, TreeLogEntry::Inserted) => {
             let leaf_index = *idx;
             *idx += 1;
             TreeInstruction::write(key, leaf_index, H256(storage_log.value.into()))
         }
-        (false, TreeLogEntry::Read { value, .. }) => {
+        (false, _, TreeLogEntry::Read { value, .. }) => {
             if storage_log.value != value {
                 tracing::error!(
-                    "Failed to map LogQuery to TreeInstruction: {:#?} != {:#?}",
-                    storage_log.value,
-                    value
-                );
-                anyhow::bail!(
-                    "Failed to map LogQuery to TreeInstruction: {:#?} != {:#?}",
+                    ?storage_log,
+                    ?tree_log_entry,
+                    "Failed to map LogQuery to TreeInstruction: read value {:#?} != {:#?}",
                     storage_log.value,
                     value
                 );
+                anyhow::bail!("Failed to map LogQuery to TreeInstruction");
             }
             TreeInstruction::Read(key)
         }
-        (false, TreeLogEntry::ReadMissingKey { .. }) => TreeInstruction::Read(key),
-        _ => {
-            tracing::error!("Failed to map LogQuery to TreeInstruction");
+        (true, StorageLogKind::RepeatedWrite, TreeLogEntry::Read { .. }) => {
```

Comment:
Is there a particular reason this match hand is treated separately from `(true, _, TreeLogEntry::Read { .. })` below? AFAIU, it uses same handling. If it were removed, the added match on `storage_log.kind` would become unnecesssary.